pkgs/games/fs2-knossos/default.nix

```diff
@@ -0,0 +1,133 @@
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, mkYarnModules
+, makeDesktopItem
+, copyDesktopItems
+, makeWrapper
+, pythonOlder
+, setuptools
+, pyqt5
+, pyqtwebengine
+, six
+, requests
+, requests-toolbelt
+, raven
+, ply
+, semantic-version
+, token-bucket
+, ninja
+, SDL2
+, openal
+, p7zip
+, nodejs
+, qt5
+, appimage-run
+}:
+
+let
+  pname = "fs2-knossos";
+  version = "0.14.3";
+
+  yarnDeps = mkYarnModules {
+    pname = "${pname}-yarn-deps";
+    inherit version;
+    packageJSON = ./package.json;
+    yarnLock = ./yarn.lock;
+    yarnNix = ./yarn.nix;
+  };
+in buildPythonApplication rec {
+  inherit pname version;
+  format = "pyproject";
+  disabled = pythonOlder "3.6";
+
+  # The rewrite is not yet in working order, but will eventually replace this.
+  # See: https://github.com/ngld/knossos
+  src = fetchFromGitHub {
+    owner = "ngld";
+    repo = "old-knossos";
+    rev = "v${version}";
+    hash = "sha256-2lTJjEzFKtFwNmqkAm3C98mX59WrAU2I/Z5ypJcA4Zk=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    ninja
+    nodejs
+    p7zip
+    qt5.qttools.dev
+    qt5.wrapQtAppsHook
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  propagatedBuildInputs = [
```

Comment:
Why does this need to be propagated?

Reply:
IIRC, everything just fails without these usable at runtime.